### PR TITLE
Shitcode the boiler into having a faster ROF depending on the amount of ammo stored

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
This makes the boiler ammo feature _NOT A COMPLETE NOOB TRAP!_
It gives some advantage and a raison to use it, instead of simply gimping xenos while adding complexity.

## Changelog
:cl:
balance: made the boiler bombard cooldown when it has more acid stored
/:cl:

### CODE REVIEW

**Comment by vilereaver41** [comment]
The "i havent played boiler ever" pr.

**Comment by KornFlaks** [comment]
More? Well the other option (#7403) just got pocket veto'ed by the maintainers.

**Comment by lbnesquik** [comment]
> 
> 
> The "i havent played boiler ever" pr. It needs more then just a simple edit like this.

I have played boiler. My second suggestion would be to simply remove the ammo mechanic, as it is currently a newbie trap.

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 1]
```diff
@@ -116,7 +117,7 @@
 
 /datum/action/xeno_action/activable/bombard/get_cooldown()
 	var/mob/living/carbon/xenomorph/boiler/X = owner
-	return X.xeno_caste.bomb_delay
+	return X.updated_delay
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:120]: this should use
`X.xeno_caste.bomb_delay - ((X.neuro_ammo + X.acid_ammo) * X.xeno_caste.ammo_multiplier)`
You wont need updated delay in that case

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm (hunk 1 above)
Comment (lbnesquik) [reply code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm]: Implemented

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 2]
```diff
@@ -233,6 +234,7 @@
 		GLOB.round_statistics.boiler_neuro_smokes++
 		SSblackbox.record_feedback("tally", "round_statistics", 1, "boiler_neuro_smokes")
 		X.neuro_ammo--
+	X.updated_delay += 10
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:237]: you don't need this then?

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm [hunk 3]
```diff
@@ -93,6 +93,7 @@
 	else
 		X.neuro_ammo++
 		to_chat(X, span_notice("We prepare a neurotoxic gas globule."))
+	X.updated_delay -= 10
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/abilities_boiler.dm:96]: ditto

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm [hunk 4]
```diff
@@ -18,7 +18,8 @@
 	//Boiler ammo
 	var/corrosive_ammo = 0
 	var/neuro_ammo = 0
-
+	///Bombard delay counter
+	var/updated_delay = 0
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm:22]: ditto

File: code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm [hunk 5]
```diff
@@ -45,7 +46,7 @@
 	ammo = GLOB.ammo_list[/datum/ammo/xeno/boiler_gas]
 	update_boiler_glow()
 	RegisterSignal(src, COMSIG_XENOMORPH_GIBBING, .proc/gib_explode)
-
+	updated_delay = xeno_caste.bomb_delay
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/boiler/boiler.dm:49]: ditto

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm [hunk 6]
```diff
@@ -141,6 +141,8 @@
 	var/bomb_strength = 0
 	///Delay between firing the bombard ability for boilers
 	var/bomb_delay = 0
+	///Used to reduce cooldown for the boiler
+	var/ammo_multiplier = 0
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:145]: weird to have this as a var IMO, but whatever

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xeno_defines.dm (hunk 6 above)
Comment (lbnesquik) [reply code/modules/mob/living/carbon/xenomorph/xeno_defines.dm:145]: Dunno what else you would want this to be, really

**Review by TiviPlus — APPROVED** [review]

**Comment by zzurge** [comment]
given that its a very nice risk reward mechanic, I think it deserves an attempt, I quite like that now we may have an incentive to use more globs instead of going with the "two shots for darkness" meta

**Comment by Nerolostshor** [comment]
true and it will give marines a reason to camp their scope. 
I will just put a but in. do not make the timer go under the time for the smoke to dissipate or a full boiler may be able so smoke lock indefinitively

**Pull request merged**